draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1137,18 +1143,72 @@ SimulationMarker = {
+annotations and concepts presented in this section.
+
+A schema defines one or more qlog event categories that can contain multiple
+event definitions. Each event MUST only belong to a single category.
 
-This document defines the main schema for the qlog format together with some
-common events, which on their own do not provide much logging utility. It is
-expected that logging is extended with specific, per-protocol event definitions
-that specify the name (category + type) and data needed for each individual
-event. Examples include the QUIC event definitions {{QLOG-QUIC}} and HTTP/3
-event definitions {{QLOG-H3}}.
+It is not possible to extend a category with new events. Instead, extension
+schema MUST define new categories with globally unique identifiers. Per
+{{{#name-field}}}, event names are the concatenation of category and type.
+Therefore, unique category identifiers ensure events in qlog files are
+disambuiguated.
+
+Each extension schema is named by a URI. That URI MUST be absolute; it precisely
+identifies the format and meaning of the extension. URIs that contain a domain
+name SHOULD also contain a month-date in the form mmyyyy. The definition of the
+schema and assignment of the URI MUST have been authorized by the owner of the
+domain name on or very close to that date. (This avoids problems when domain
+names change ownership.) If the resource or document defines several event
+categories, then the URI MUST identify the actual category in use, e.g., using
+a fragment or query identifier (characters after a "#" or "?" in the URI).
+
+For extensions defined in RFCs, the URI used SHOULD be a URN starting with
+`urn:ietf:params:qlog:` followed by a registered, descriptive name.
+
+A log file that uses events from extension schema SHOULD list all schema
+identifiers in the `additional_event_schemas` field; see {{qlog-file-schema}}
+and {{qlog-file-seq-schema}}.
+
+In the following example, a qlog file contains two hypothetical standardized
+extension schema, along with a private extension. The first schema is named
+"rick" and consists of categories "roll", "astley", and "moranis". The second
+schema is named "john" and consists of categories "doe", "candy", and "goodman".
+The private schema is named "pickle" and consists of categories "pepper",
+"lilly", and "rick":
+
+~~~
+additional_event_schemas = [
+                            urn:ietf:params:qlog:rick#roll,
+                            urn:ietf:params:qlog:rick#astley,
+                            urn:ietf:params:qlog:rick#moranis,
+                            urn:ietf:params:qlog:john#doe,
+                            urn:ietf:params:qlog:john#candy,
+                            urn:ietf:params:qlog:john#goodman,
+                            https://example.com/032024/pickle.html#pepper,
+                            https://example.com/032024/pickle.html#lilly,
+                            https://example.com/032024/pickle.html#rick,
+                          ]
```

Comment:
```suggestion
"additional_event_schemas": [
                            "urn:ietf:params:qlog:rick#roll",
                            "urn:ietf:params:qlog:rick#astley",
                            "urn:ietf:params:qlog:rick#moranis",
                            "urn:ietf:params:qlog:john#doe",
                            "urn:ietf:params:qlog:john#candy",
                            "urn:ietf:params:qlog:john#goodman",
                            "https://example.com/032024/pickle.html#pepper",
                            "https://example.com/032024/pickle.html#lilly",
                            "https://example.com/032024/pickle.html#rick",
                          ]
```

Since it's an example of a JSON serialized qlog file, it should be valid JSON.